tests/test_cmdline.py

```diff
@@ -0,0 +1,62 @@
+import zipline.__main__ as main
+import zipline
+from zipline.testing import ZiplineTestCase
+from zipline.testing.predicates import (
+    assert_equal,
+    assert_raises_str,
+)
+from click.testing import CliRunner
+from zipline.extensions import ExtensionArgs
+
+
+class CmdLineTestCase(ZiplineTestCase):
+
+    def init_instance_fixtures(self):
+        super(CmdLineTestCase, self).init_instance_fixtures()
+
+        # make sure this starts empty
+        # TODO: decide how this needs to be reset
+        zipline.extension_args = ExtensionArgs([])
+
+    def test_parse_args(self):
```

Comment:
we should add some tests for nested keys, keys that have numbers in them, keys with underscores, and any other interesting names you can think of.

Reply:
I'll add some tests here for weird names, but currently I keep all of the tests for nested vs non-nested separate (no particular reason, just the way I had started doing it earlier), so I'll add nested ones to `test_parse_namespaces()`.